src/bin/collect_stats.rs

```diff
@@ -0,0 +1,88 @@
+use ed25519_dalek::{SecretKey, SigningKey};
+use hex;
+use snapchain::proto::snapchain::Block;
+use std::time::Duration;
+use submit_message::compose_message;
+use tokio::sync::mpsc;
+use tokio::{select, time};
+mod submit_message;
+use hex::FromHex;
+mod follow_blocks;
+use message::MessageData;
+use prost::Message;
+use snapchain::proto::message;
+// Sumbit x messages per second
+// Read blocks and collect stats
+const SUBMIT_MESSAGE_INTERVAL: Duration = Duration::from_millis(100);
+const STATS_CALCULATION_INTERVAL: Duration = Duration::from_secs(1);
+const RPC_ADDR: &str = "http://127.0.0.1:3383";
+
+#[tokio::main]
+async fn main() {
+    // feel free to specify your own key
+    let (blocks_tx, mut blocks_rx) = mpsc::channel::<Block>(1000);
+    let (messages_tx, mut messages_rx) = mpsc::channel::<message::Message>(1000);
+    tokio::spawn(async move {
+        let private_key = SigningKey::from_bytes(
+            &SecretKey::from_hex(
+                "1000000000000000000000000000000000000000000000000000000000000000",
+            )
+            .unwrap(),
+        );
+        // Create a timer for block creation
+        let mut submit_message_timer = time::interval(SUBMIT_MESSAGE_INTERVAL);
+
+        loop {
+            let _ = submit_message_timer.tick().await;
+            let message = compose_message(
+                private_key.clone(),
+                6833,
+                RPC_ADDR.to_string(),
+                "For benchmarking",
```

Comment:
Maybe we can append an incrementing number each message? When we start actually hashing these messages we probably want to ensure they hash to different values (although hashing to the same value would be a different kind of test).

Reply:
Sounds good, made the change.